.github/workflows/workflow.yml

```diff
@@ -73,34 +73,49 @@ jobs:
     - name: Initialisation
       uses: actions/checkout@v2
 
-    - name: Install 
+    - name: Create conda env
+      run: conda create -n test-env -c conda-forge -y python==3.8 pip
+
+    - name: Install packages
       run: |
+        eval "$(conda shell.bash hook)"
+        conda activate test-env
+        python --version
         python -m pip install invoke
         python -m pip install git+https://github.com/ubermag/ubermagutil.git
         python -m pip install git+https://github.com/ubermag/ubermagtable.git
         python -m pip install git+https://github.com/ubermag/discretisedfield.git
         python -m pip install git+https://github.com/ubermag/micromagneticmodel.git
         python -m pip install git+https://github.com/ubermag/micromagnetictests.git
+        python -m pip install .[dev]
 
-    - name: Install package
-      run: python -m pip install .[dev]
-
-    - name: Unit tests
-      run: python -m pytest -v mumax3c
+    - name: Test mumax3c
+      run: |
+        eval "$(conda shell.bash hook)"
+        conda activate test-env
+        python -m pytest -v mumax3c
```

Comment:
We should switch back to using `invoke` here and I would suggest to test coverage on the Linux runner.

Reply:
I have made the changes, have a look. Do you think we should add `if: always()` to test coverage even if other tests fail?

Reply:
@lang-m I understand. I have made the required changes, have a look.

Reply:
Looks good